File: Easy leetcode/1250. Check If It Is a Good Array.py
import math

class Solution:
    def isGoodArray(self, nums):
        g = nums[0]
        for i in range(1, len(nums)):
            g = math.gcd(g, nums[i])
            if g == 1:
                return True
        return g == 1  # g==1 hole true jabe na hole flase jabe

File: Easy leetcode/1250. Check If It Is a Good Array.java
import java.util.*;

class Solution {
    public boolean isGoodArray(int[] nums) {
        int g = nums[0];
        for (int i = 1; i < nums.length; i++) {
            g = gcd(g, nums[i]);
            if (g == 1) {
                return true;
            }
        }
        return g == 1; // g==1 hole true jabe na hole flase jabe
    }

    private int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}

